cloud/azure/azure_rm_deployment.py

```diff
@@ -0,0 +1,644 @@
+        '''
+        Return True/False based on existence of requested resource group.
+
+        :param resource_group: string. Name of a resource group.
+        :return: boolean
+        '''
+        try:
+            self.rm_client.resource_groups.get(resource_group)
+        except CloudError:
+            return False
+        return True
+
+    def _get_failed_nested_operations(self, current_operations):
+        new_operations = []
+        for operation in current_operations:
+            if operation.properties.provisioning_state == 'Failed':
+                new_operations.append(operation)
+                if operation.properties.target_resource and \
+                   'Microsoft.Resources/deployments' in operation.properties.target_resource.id:
+                    nested_deployment = operation.properties.target_resource.resource_name
+                    try:
+                        nested_operations = self.rm_client.deployment_operations.list(self.resource_group_name,
+                                                                                      nested_deployment)
+                    except CloudError as exc:
+                        self.fail("List nested deployment operations failed with status code: %s and message: %s" %
+                                 (e.status_code, e.message))
+                    new_nested_operations = self._get_failed_nested_operations(nested_operations)
+                    new_operations += new_nested_operations
+        return new_operations
+
+    def _get_failed_deployment_operations(self, deployment_name):
+        results = []
+        # time.sleep(15) # there is a race condition between when we ask for deployment status and when the
+        #               # status is available.
+
+        try:
+            operations = self.rm_client.deployment_operations.list(self.resource_group_name, deployment_name)
+        except CloudError as exc:
+            self.fail("Get deployment failed with status code: %s and message: %s" %
+                      (exc.status_code, exc.message))
+        try:
+            results = [
+                dict(
+                    id=op.id,
+                    operation_id=op.operation_id,
+                    status_code=op.properties.status_code,
+                    status_message=op.properties.status_message,
+                    target_resource=dict(
+                        id=op.properties.target_resource.id,
+                        resource_name=op.properties.target_resource.resource_name,
+                        resource_type=op.properties.target_resource.resource_type
+                    ) if op.properties.target_resource else None,
+                    provisioning_state=op.properties.provisioning_state,
+                )
+                for op in self._get_failed_nested_operations(operations)
+            ]
+        except:
+            # If we fail here, the original error gets lost and user receives wrong error message/stacktrace
```

Comment:
should probably at least try to log this failure (to our neutered log, but still)